File: caper/samples/ruby/hello0.rb
require './hello0_parser.rb'

class SemanticAction
    def syntax_error
    end

    def stack_overflow
    end
    
    def downcast v
        v
    end

    def upcast v
        v
    end
end

parser = Hello_world::Parser.new(SemanticAction.new)

parser.post :token_Hello, 0
parser.post :token_World, 0

File: caper/samples/php/hello0.php
<?php

error_reporting(E_ALL);

require_once('hello0_parser.php');

class SemanticAction
{
    function syntax_error()
    {
    }

    function stack_overflow()
    {
    }
    
    function downcast($y)
    {
        return $y;
    }

    function upcast($y)
    {
        return $y;
    }
}

$parser = new hello_world\Parser(new SemanticAction());

$parser->post(hello_world\Token::token_Hello, 0);
$parser->post(hello_world\Token::token_World, 0);
